src/adapters/fetch.ts

```diff
@@ -10,12 +10,12 @@ export async function cachedFetchComponent(
   const logger = logs.getLogger('cached-fetch-component')
 
   function cache<T extends object>() {
-    return new LRUCache<string, T>({
+    const lruCache = new LRUCache<string, T>({
       max: 1000,
       ttl: 1000 * 60 * 5, // 5 min,
       fetchMethod: async function (url: string, _staleValue: T | void): Promise<T> {
         try {
-          const response = await fetch.fetch(url)
+          const response = await fetch.fetch(url, { method: 'GET' })
```

Comment:
It seems that the concept of a cached fetch using the LRUCache dependency has reached it's limit, as it can only receive a string as the cache key. You could try and hack it so it serializes the body data as part of the cache key, but it might be better to either:
1. Use another dependency that is made for caching fetch requests.
2. Create a new component called cached that uses the LRU cache and lets you define caches based on a key that's not the URL. You could for example, use it in the new names component to cache the whole process (instead of just the fetch), and use as the keys, the addresses that you were asked to retrieve. Although they need to be serialized and have a common order to be sure you'll match it.
3. Not cache requests by a body.